hydroflow/src/util/mod.rs

```diff
@@ -250,3 +250,51 @@ mod test {
         );
     }
 }
+
+use std::io::Read;
+use std::process::{Child, ChildStdin, ChildStdout, Stdio};
+
+/// TODO:
+pub fn wait_for_output(output_so_far: &mut String, output: &mut ChildStdout, wait_for: &str) {
+    let re = regex::Regex::new(wait_for).unwrap();
+
+    while !re.is_match(output_so_far) {
+        println!("waiting: {}", output_so_far);
+        let mut buffer = [0u8; 1024];
+        let bytes_read = output.read(&mut buffer).unwrap();
+
+        if bytes_read == 0 {
+            panic!();
+        }
+
+        output_so_far.push_str(&String::from_utf8_lossy(&buffer[0..bytes_read]));
```

Comment:
Probaby want to use read_to_string https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_string

Reply:
read_to_string blocks until EOF which is not what we want here